queue/priority_queue.go

```diff
@@ -253,6 +258,12 @@ func (pq *PriorityQueue) Dispose() {
 	pq.waiters = nil
 }
 
+// AllowDuplicates determines whether the queue supports
+// duplicated items. This setting is false by default.
+func (pq *PriorityQueue) AllowDuplicates(allow bool) {
+	pq.allowDuplicates = allow
```

Comment:
I know it's a bigger refactor for consumers, but this probably makes sense as a constructor argument.  I doubt anyone is going to change this after instantiation.